nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/smoother_selector_node.hpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Pablo Iñigo Blasco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__SMOOTHER_SELECTOR_NODE_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__SMOOTHER_SELECTOR_NODE_HPP_
+
+#include <memory>
+#include <string>
+
+#include "std_msgs/msg/string.hpp"
+
+#include "behaviortree_cpp_v3/action_node.h"
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_behavior_tree
+{
+
+/**
+ * @brief The SmootherSelector behavior is used to switch the smoother
+ * that will be used by the smoother server. It subscribes to a topic "smoother_selector"
+ * to get the decision about what smoother must be used. It is usually used before of
+ * the FollowPath. The selected_smoother output port is passed to smoother_id
+ * input port of the FollowPath
+ */
+class SmootherSelector : public BT::SyncActionNode
```

Comment:
Make sure to open a PR for documentation updates: migration guide mentioning this exists, add to the list of navigation plugins, and a configuration guide page for the parameters